smoke/IotEdgeQuickstart/Program.cs

```diff
@@ -287,6 +290,7 @@ async Task<int> OnExecuteAsync()
                     tag,
                     this.DeviceId,
                     this.EdgeHostname,
+                    string.IsNullOrWhiteSpace(this.ParentHostname) ? Option.None<string>() : Option.Some(this.ParentHostname),
```

Comment:
Do we need this `IsNullOrWhiteSpace()` here? Since on line178, we already have the default value set to empty. Do we need the type to be `Option<string>` instead of just plain `string`?

Reply:
To use IsNullOrWhiteSpace(), it is defensive to protect if white spaces are passing in.  For Option<string>, all optional string argument should use it to show intent explicitly if the argument is not provided.